Index: glibc-2.28/sysdeps/hurd/include/hurd/signal.h
===================================================================
--- glibc-2.28.orig/sysdeps/hurd/include/hurd/signal.h
+++ glibc-2.28/sysdeps/hurd/include/hurd/signal.h
@@ -11,6 +11,11 @@ libc_hidden_proto (_hurd_exception2signa
 libc_hidden_proto (_hurd_intr_rpc_mach_msg)
 libc_hidden_proto (_hurd_thread_sigstate)
 libc_hidden_proto (_hurd_raise_signal)
+libc_hidden_proto (_hurd_sigstate_set_global_rcv)
+libc_hidden_proto (_hurd_sigstate_lock)
+libc_hidden_proto (_hurd_sigstate_pending)
+libc_hidden_proto (_hurd_sigstate_unlock)
+libc_hidden_proto (_hurd_sigstate_delete)
 #endif
 #ifdef _HURD_SIGNAL_H_HIDDEN_DEF
 libc_hidden_def (_hurd_self_sigstate)
Index: glibc-2.28/hurd/hurdsig.c
===================================================================
--- glibc-2.28.orig/hurd/hurdsig.c
+++ glibc-2.28/hurd/hurdsig.c
@@ -171,6 +171,7 @@ __hurd_sigstate_delete_2_13 (thread_t th
 }
 compat_symbol (libc, __hurd_sigstate_delete_2_13, _hurd_sigstate_delete, GLIBC_2_13_DEBIAN_19);
 #endif
+libc_hidden_ver (__hurd_sigstate_delete, _hurd_sigstate_delete)
 
 /* Make SS a global receiver, with pthread signal semantics.  */
 void
@@ -188,6 +189,7 @@ __hurd_sigstate_set_global_rcv_2_13 (str
 }
 compat_symbol (libc, __hurd_sigstate_set_global_rcv_2_13, _hurd_sigstate_set_global_rcv, GLIBC_2_13_DEBIAN_19);
 #endif
+libc_hidden_ver (__hurd_sigstate_set_global_rcv, _hurd_sigstate_set_global_rcv)
 
 /* Check whether SS is a global receiver.  */
 static int
@@ -214,7 +216,9 @@ __hurd_sigstate_unlock (struct hurd_sigs
     __spin_unlock (&_hurd_global_sigstate->lock);
 }
 versioned_symbol (libc, __hurd_sigstate_lock, _hurd_sigstate_lock, GLIBC_2_21);
+libc_hidden_ver (__hurd_sigstate_lock, _hurd_sigstate_lock)
 versioned_symbol (libc, __hurd_sigstate_unlock, _hurd_sigstate_unlock, GLIBC_2_21);
+libc_hidden_ver (__hurd_sigstate_unlock, _hurd_sigstate_unlock)
 
 #if SHLIB_COMPAT (libc, GLIBC_2_13, GLIBC_2_21)
 void
@@ -250,6 +254,7 @@ __hurd_sigstate_pending_2_13 (const stru
 }
 compat_symbol (libc, __hurd_sigstate_pending_2_13, _hurd_sigstate_pending, GLIBC_2_13_DEBIAN_19);
 #endif
+libc_hidden_ver (__hurd_sigstate_pending, _hurd_sigstate_pending)
 
 /* Clear a pending signal and return the associated detailed
    signal information. SS must be locked, and must have signal SIGNO
