---
 localedata/locales/sv_SE |    4 ++++
 1 file changed, 4 insertions(+)

--- a/localedata/locales/sv_SE
+++ b/localedata/locales/sv_SE
@@ -200,6 +200,10 @@
 t_fmt     "%H:%M:%S"
 am_pm     "";""
 t_fmt_ampm  ""
+% Appropriate date representation (date(1))   "%a %e %b %Y %H:%M:%S %Z"
+date_fmt       "<U0025><U0061><U0020><U0025><U0065><U0020><U0025><U0062>/
+<U0020><U0025><U0059><U0020><U0025><U0048><U003A><U0025><U004D><U003A>/
+<U0025><U0053><U0020><U0025><U005A>"
 week 7;19971130;4
 first_weekday 2
 END LC_TIME
